### PR TITLE
Feat/book card list component

### PR DESCRIPTION
**Pull request by tranchinh123**

### CODE REVIEW

File: bookshelter-app/src/components/BookCardList/index.tsx [hunk 1]
```diff
@@ -1,5 +1,40 @@
...
+
 const BookCardList = () => {
-  return <></>;
+  const [books, setBooks] = useState<Books[]>([]);
+
+  useEffect(() => {
+    const loadBookList = async (): Promise<void> => {
+      const fetchBooks = await get<Books[]>(API.BOOKS_ENDPOINT);
+      if (fetchBooks) {
+        console.log(fetchBooks);
```
Comment (huypham-agilityio) [inline bookshelter-app/src/components/BookCardList/index.tsx]: remove log

File: bookshelter-app/src/components/BookCardList/index.tsx [hunk 2]
```diff
@@ -1,5 +1,40 @@
...
+
+        setBooks(fetchBooks);
+      }
+    };
+    loadBookList();
+  }, []);
+
+  return (
+    <>
```
Comment (huypham-agilityio) [inline bookshelter-app/src/components/BookCardList/index.tsx]: do we need <></>?

File: bookshelter-app/src/components/BookCardList/index.tsx [hunk 3]
```diff
@@ -1,5 +1,40 @@
...
+import { Books } from '../../types/index';
+import { API } from '../../constants/api';
+
 const BookCardList = () => {
-  return <></>;
+  const [books, setBooks] = useState<Books[]>([]);
+
+  useEffect(() => {
+    const loadBookList = async (): Promise<void> => {
+      const fetchBooks = await get<Books[]>(API.BOOKS_ENDPOINT);
```
Comment (huypham-agilityio) [inline bookshelter-app/src/components/BookCardList/index.tsx]: update naming

File: bookshelter-app/src/components/BookCardList/index.tsx (hunk 1 above)
Comment (tranchinh123) [reply bookshelter-app/src/components/BookCardList/index.tsx]: Fixed!

File: bookshelter-app/src/components/BookCardList/index.tsx (hunk 2 above)
Comment (tranchinh123) [reply bookshelter-app/src/components/BookCardList/index.tsx]: Fixed!

File: bookshelter-app/src/components/BookCardList/index.tsx (hunk 3 above)
Comment (tranchinh123) [reply bookshelter-app/src/components/BookCardList/index.tsx]: Fixed!

File: bookshelter-app/src/components/BookCardList/index.tsx [hunk 4]
```diff
@@ -1,5 +1,36 @@
+import BookCard from '../BookCard';
+import styles from './index.module.css';
+import { get } from '../../services/api';
+import { useState, useEffect } from 'react';
+import { Books } from '../../types/index';
+import { API } from '../../constants/api';
+
 const BookCardList = () => {
```
Comment (huypham-agilityio) [inline bookshelter-app/src/components/BookCardList/index.tsx]: books should be passed via prop

Comment (tranchinh123) [reply bookshelter-app/src/components/BookCardList/index.tsx]: Fixed!

File: bookshelter-app/src/components/BookCardList/index.tsx [hunk 5]
```diff
@@ -1,5 +1,31 @@
-const BookCardList = () => {
-  return <></>;
+import BookCard from '../BookCard';
+import styles from './index.module.css';
+
+interface book {
```
Comment (huypham-agilityio) [inline bookshelter-app/src/components/BookCardList/index.tsx:4]: Book

Comment (tranchinh123) [reply bookshelter-app/src/components/BookCardList/index.tsx:4]: Fixed!

**Pull request merged**